src/vs/vscode.proposed.d.ts

```diff
@@ -1956,6 +1956,116 @@ declare module 'vscode' {
 
 	//#endregion
 
+	//@region https://github.com/microsoft/vscode/issues/16221
+
+	export namespace languages {
+		/**
+		 * Register a inline hints provider.
+		 *
+		 * Multiple providers can be registered for a language. In that case providers are sorted
+		 * by their [score](#languages.match) and the best-matching provider is used. Failure
+		 * of the selected provider will cause a failure of the whole operation.
+		 *
+		 * @param selector A selector that defines the documents this provider is applicable to.
+		 * @param provider An on type inline hints provider.
+		 * @return A [disposable](#Disposable) that unregisters this provider when being disposed.
+		 */
+		export function registerInlineHintsProvider(selector: DocumentSelector, provider: InlineHintsProvider): Disposable;
+	}
+
+	export interface ThemableDecorationAttachmentRenderOptions {
+		/**
+		 * CSS styling property that will be applied to the decoration attachment.
+		 */
+		fontSize?: string;
+		/**
+		 * CSS styling property that will be applied to the decoration attachment.
+		 */
+		fontFamily?: string;
+		/**
+		 * CSS styling property that will be applied to the decoration attachment.
+		 */
+		padding?: string;
+	}
+
+	/**
+	 * Inline hint information.
+	 */
+	export class InlineHint {
+		/**
+		 * The text of the hint.
+		 */
+		text: string;
+		/**
+		 * The range of the hint.
+		 */
+		range: Range;
+		/**
+		 * The position of hint trigger.
+		 */
+		triggerPosition: Position;
+		/**
+		 * Prefix of hint text.
+		 */
+		prefix?: string;
+		/**
+		 * Postfix of hint text.
+		 */
+		postfix?: string;
+		/**
+		 * Context for the hint.
+		 */
+		contextValue?: string;
```

Comment:
Sorry, I am not up to speed with all these changes but I am not a fan of those four fields (triggerPosition, prefix, postfix, contextValue)

Reply:
The information returned by the API should be self-contained and not rely on something from the outside, like context menus. The language API should also have a future and LSP and there such concepts (context value etc) do not exist